src/javascripts/Services/gridService.js

```diff
@@ -0,0 +1,90 @@
+        // scope functions
+        scope.reload = function() {
+            if (!scope.tree) return;
+            let oldTree = scope.tree;
+            scope.clearSelection(true);
+            scope.buildTree();
+            scope.releaseTree(oldTree);
+        };
+
+        scope.resolveNodeError = (path, part) => {
+            return new Error(`Failed to resolve node "${part}" in path "${path}"`);
+        };
+
+        scope.onNodeMouseDown = function(e, node) {
+            if (e.button !== 2 || !node.selected) scope.selectNode(e, node);
+            if (e.button === 2) scope.showContextMenu(e);
+        };
+
+        scope.resolveElements = function() {
+            scope.tabView = element[0];
+            scope.treeElement = htmlHelpers.resolveElement(scope.tabView, '.tree-nodes');
+            scope.columnsElement = htmlHelpers.resolveElement(scope.tabView, '.column-wrapper');
+        };
+        
+        scope.buildColumns = function() {
+            scope.columns = scope.allColumns.filter(function(column) {
+                return column.enabled;
+            });
+            let width = scope.columns.reduce(function(width, c) {
+                if (c.width) width += parseInt(c.width.slice(0, -1));
+                return width;
+            }, 0);
+            if (width > 100) {
+                let defaultWidth = Math.floor(100 / scope.columns.length) + '%';
+                scope.columns.slice(0, -1).forEach(column => {
+                    column.width = defaultWidth
+                });
+            }
+            scope.resizeColumns();
+        };
+
+        scope.resolveNode = function(path) {
+            let node,
+                handle = 0;
+            path.split('\\').forEach(function(part) {
+                let nextHandle = xelib.GetElementEx(handle, `${part}`);
+                if (handle > 0) xelib.Release(handle);
+                handle = nextHandle;
+                if (part !== 'Child Group') {
+                    node = scope.getNodeForElement(handle);
+                    if (!node) throw scope.resolveNodeError(path, part);
+                    if (!node.has_data) scope.getNodeData(node);
+                }
+            });
+            return node;
+        };
+
+        scope.buildColumnValues = function(node) {
+            node.column_values = scope.columns.map(function(column) {
+                try {
+                    return column.getData(node, xelib);
+                } catch (x) {
+                    console.log(x);
+                    return { value: '' };
+                }
+            }).trimFalsy();
+        };
+
+        scope.getNodeData = function(node) {
+            node.has_data = true;
+            node.fid = xelib.GetFormID(node.handle);
+            scope.buildColumnValues(node);
+        };
+
+        scope.getNodeClass = function(node) {
+            node.class = ' ';
+        };
+
+        scope.buildNode = function(handle) {
+            return {
```

Comment:
I also do not understand this function.  it seems very simplistic.

I do know I had a buildNode function for the tree view and the record view, and I can see how it might be this simple for a grid view, but I'm still curious.